pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -522,16 +522,30 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, nodeName string) (diskID string
 
 	err = vm.AddDevice(ctx, disk)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	vmDevices, err = vm.Device(ctx)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	devices := vmDevices.SelectByType(disk)
-	return devices.Name(devices[len(devices)-1]), nil
+	newDevice := devices[len(devices)-1]
+
+	deviceUuid := getVirtualDiskUuid(newDevice)
```

Comment:
no need to declare a new variable here, can use 'diskUUID' which is already defined in your return signature for this function